pkg/cloudprovider/aws/cloudprovider.go

```diff
@@ -205,3 +235,39 @@ func (c *CloudProvider) getProvider(ctx context.Context, provider *runtime.RawEx
 	}
 	return aws, nil
 }
+
+func (c *CloudProvider) useOpinionatedInstanceFilter(provisionerRequirements ...v1.NodeSelectorRequirement) bool {
+	var instanceTypeRequirement, familyRequirement v1.NodeSelectorRequirement
```

Comment:
optional: Check out the scheduling.Requirements library to potentially remove a bunch of undifferentiated code. 
```
requirements := scheduling.NewNodeSelectorRequirements(provisionerRequirements)
for _, label := range string[]{ v1.LabelInstanceTypeStable, v1alpha1.LabelInstanceFamily } {
  if requirements.Has(label) && requirements.Get(label).Operator != v1.NodeSelectorOpIn {
    return false
  }
}
return true
```